src/utils/channel_utils.js

```diff
@@ -546,6 +546,16 @@ export function sortChannelsByDisplayNameAndMuted(locale, members, a, b) {
     return -1;
 }
 
+export function sortChannelsByRecency(lastPosts, a, b) {
+    const aLastPostAt = (lastPosts[a.id] && lastPosts[a.id].update_at) || a.last_post_at;
```

Comment:
I'm not sure I understand the logic here. Is this supposed to be equivalent to `lastPosts[a.id] ? lastPosts[a.id].update_at : a.last_post_at`

Reply:
Yes, that's the idea.

If a post exists for that channel get the `updated_at` timestamp.
If not get the timestamp of the channel property `last_post_at `

I noticed that `updated_at` is not updated if a new post is added to the channel.